cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/GlobalParam.scala

```diff
@@ -0,0 +1,213 @@
+package com.microsoft.azure.synapse.ml.services
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+
+sealed trait GlobalKey[T]
+sealed trait GlobalParamKey[T] extends GlobalKey[T]
+sealed trait GlobalServiceParamKey[T] extends GlobalKey[Either[T, String]] //left is Scalar, right is Vector
+
+case object OpenAIDeploymentNameKey extends GlobalServiceParamKey[String]
```

Comment:
can you put this where OpenAI is definred so that people can see how to hook into you API

Reply:
Yes!